### PR TITLE
Create gui

### PR DESCRIPTION
**Pull request by jelledouwe**

### CODE REVIEW

File: eagerx_core/src/eagerx_core/gui/rxgui_node.py [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+        if key == 'color':
+            self.node.graphics_item().set_color()
+            self.node.graphics_item().update()
+
+    def value_changed(self, widget, key):
+        self.node.set_param(key, widget.value())
+
+    def text_changed(self, text, key):
+        try:
+            self.node.set_param(key, yaml.safe_load(str(text)))
```
Comment (bheijden) [inline eagerx_core/src/eagerx_core/gui/rxgui_node.py:536]: yaml not imported. This statement will always result in an exception that is subsequently not raised.

**Review by bheijden — CHANGES_REQUESTED** [review]
- Only allow a single parameter window open & block the rest of the windows. I keep losing track of open windows.
- Closing converter window pushes the parameter window behind the graph window --> As such, losing this window out of my sight.
- update todos in __init__ (or create github issues)
- env/actions & env/observations: hide "rate", fix "process". (hoverinfo & parameter screen)
- observation terminal: hide "is_reactive", "rate", "space_converter". (hoverinfo & parameter screen)
- action terminal: hide "start_with_msg", "space_converter". (hoverinfo & parameter screen)
- state terminal: hide "converter". (hoverinfo & parameter screen)
- target terminal: hide space_converter. (hoverinfo & parameter screen)
- sensor terminal: hide "start_with_msg", because always False.  (hoverinfo & parameter screen)
- Crash (i.e. assertion not gracefully handled) if changing the converter of an observation to identity (i.e. not a spaceconverter).
- No msg_type error / disconnect after changing converter of obs_2 from "SpaceUInt64Converter" to "SpaceStringConverter", while that should happen. Some thoughts: 
 -- Changing converter of obs_2 --> converter_old in rxgraph._set_converter is already set to new converter (see for yourself with a breakpoint in rxgraph._set_converter)
-- Seems that you modify the converter manually somewhere in your code before calling rxgraph.set_parameter('converter', converter, ..)
- Crash: Open parameters of feedthrough "out", then edit IdentityConverter, change nothing and click "x". results in crash "Invalid Action: msg_type'
- Basically, converter msg_type check is no where performed. Seems related to previous comment.

File: eagerx_core/src/eagerx_core/gui/rxgui_node.py (hunk 1 above)
Comment (bheijden) [reply eagerx_core/src/eagerx_core/gui/rxgui_node.py:536]: - Feedthrough terminal: show (fixed) msg_type & space_converter of corresponding output. (hoverinfo, parameter window)

**Comment by bheijden** [comment]
Feedthrough terminal: show (fixed) msg_type & space_converter of corresponding output. (hoverinfo, parameter window)

**Comment by bheijden** [comment]
- reload _state after set_params(..), as connections might be removed (if e.g. the edited converter changes the msg type)

File: eagerx_core/src/eagerx_core/gui/__init__.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# todo: Check converter validity before connecting
+# todo: Check if flowchart can run in notebook
+# todo: Add render node
+# todo: What to do if text is invalid yaml?
+# todo: rate should not be none (now happens when connecting an observation)
+# todo: Nice to have: show rate in outputs
+# todo: Nice to have: allow to add terminals in nodes
+# todo: Nice to have: Input terminals get address on connection
+# todo: Nice to have: Generate rate based on connection (not changeable)
+# todo: Nice to have: Show addresses and update on changes in connection (not changeable)
```
Comment (bheijden) [inline eagerx_core/src/eagerx_core/gui/__init__.py:39]: Move these to github as issues if they are long-term. Add a tag "gui".

**Review by bheijden — COMMENTED** [review]
Very nice! Gui seems solid now!

Some final remarks:
- action/observation node parameter dialog has process="1" instead of "environment"
- Connecting an action with an observation gives a strange error: "argument of type 'type' is not iterable" --> Make a more informative assertion error message?
- Suppress printing the AssertionErrors in console that were caught by pyqt. These errors will already be shown in a dialog box anyway, right? Avoids clutter in console.
- In the node selection menu, where we can select nodes to be added, can you filter all nodes that have "tick" as "inputs" or "outputs". These are simulation nodes/bridges that may not be added!

Other than that, I think it is ready to be merged in!

**Comment by bheijden** [comment]
- Connections window that opens if I connect an action shows "identityConverter" while it should show the SpaceConverter of the target.
-  When I removed an object, it did not clear the parameters of the disconnected observations
